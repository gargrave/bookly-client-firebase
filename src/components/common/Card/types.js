// @flow
export type CardTextListItem = {
  title?: string,
  value?: string,
}
