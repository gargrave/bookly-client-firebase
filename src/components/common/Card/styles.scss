@import '../../../styles/_index';

.bookly-card {
  @include boxShadowMed;
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  margin: 8px auto;
  max-width: 600px;
  padding: 10px 20px;

  &--top-margin-med {
    margin-top: 50px;
  }

  hr {
    margin: 12px 0;
  }

  &--hoverable:hover {
    background: #e4effc;
    cursor: pointer;
  }

  &__header,
  &__title,
  &__text {
    margin: 0;
  }

  &__header,
  &__title {
    font-weight: bold;
  }

  &__header {
    font-size: 1.5rem;
  }

  &__text {
    color: #6e6969;
  }

  &__text + &__text {
    margin-top: 8px;
  }
}

.bookly-detail-card {
  padding: 15px;

  .bookly-card__title {
    font-size: 1.3rem;
    line-height: 2.5rem;
  }

  .bookly-card__title + .bookly-card__text {
    font-size: 1.1rem;
  }
}

.bookly {
  &-list-detail-card {
    padding: 10px
  }
}
