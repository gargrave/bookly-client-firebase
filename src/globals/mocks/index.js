import { authorMocks } from './data/authorMocks';
import { bookBucketMocks } from './data/bookBucketMocks';
import { bookMocks } from './data/bookMocks';
import { profileMocks } from './data/profileMocks';
import { userMocks } from './data/userMocks';

export {
  authorMocks,
  bookBucketMocks,
  bookMocks,
  profileMocks,
  userMocks,
};
