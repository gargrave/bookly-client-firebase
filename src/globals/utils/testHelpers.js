import React from 'react';
import { shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = defaultProps;
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }
}
