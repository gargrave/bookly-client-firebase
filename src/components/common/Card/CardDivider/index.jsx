// @flow
import React from 'react';

import { buildClasses } from '../../../../utils/cssHelpers';

type Props = {
};

const CardDivider = ({}: Props) => {
  return (
    <hr className={buildClasses(['card__divider'])} />
  );
};

CardDivider.propTypes = {
};

export default CardDivider;
