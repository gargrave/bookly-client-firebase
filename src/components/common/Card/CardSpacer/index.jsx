// @flow
import React from 'react';
import { oneOf } from 'prop-types';

import { buildClasses } from '../../../../utils/cssHelpers';

import './styles.css';

type Props = {
  size?: string,
};

const CardSpacer = ({
  size = 'medium',
}: Props) => {
  return (
    <div className={buildClasses(`card__spacer--${size}`)} />
  );
};

CardSpacer.propTypes = {
  size: oneOf(['small', 'medium', 'large']),
};

export default CardSpacer;
