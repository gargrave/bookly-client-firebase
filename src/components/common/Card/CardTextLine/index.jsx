// @flow
import React from 'react';
import { bool, string } from 'prop-types';

import { buildClasses } from '../../../../utils/cssHelpers';

type Props = {
  bold?: boolean,
  condensed?: boolean,
  text: string,
};

const CardTextLine = ({
  bold = false,
  condensed = false,
  text,
}: Props) => {
  return (
    <p
      className={buildClasses([
        'card__text',
        bold ? 'card__text--bold' : '',
        condensed ? 'condensed' : '',
      ])}
    >
      {text}
    </p>
  );
};

CardTextLine.propTypes = {
  bold: bool,
  condensed: bool,
  text: string.isRequired,
};

export default CardTextLine;
