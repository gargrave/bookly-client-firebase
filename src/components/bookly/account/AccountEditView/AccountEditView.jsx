// @flow
import React from 'react';
import { object } from 'prop-types';

import type { Profile, User } from '../../../../globals/flowtypes';

import { buildClasses } from '../../../../globals/utils/cssHelpers';
import profileModel from '../../../../models/Profile.model';

import Card from '../../../common/Card/Card';

import AccountForm from '../AccountForm/AccountForm';

type Props = {
  profile: Profile,
  user: User,
};

const AccountEditView = ({
  profile,
  user,
}: Props) => {
  return (
    <div className={buildClasses(['detail-view', 'account-edit-view'])}>
      <Card
        classes={['detail-card', 'account-edit-card']}
        header="Update My Account"
        hoverable={false}
      >
        <AccountForm
          disabled={false}
          errors={profileModel.emptyErrors()}
          onCancel={() => {}}
          onInputChange={() => {}}
          onSubmit={() => {}}
          profile={profile}
          submitDisabled={false}
          topLevelError={''}
        />
      </Card>
    </div>
  );
};

AccountEditView.propTypes = {
  profile: object.isRequired,
  user: object.isRequired,
};

export default AccountEditView;
