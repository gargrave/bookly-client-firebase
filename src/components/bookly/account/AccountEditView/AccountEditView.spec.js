import { profileMocks, userMocks } from '../../../../globals/mocks/';
import { ComponentBuilder } from '../../../../globals/utils/testHelpers';

import Card from '../../../common/Card/Card';

import AccountEditView from './AccountEditView';

const defaultProps = {
  profile: { ...profileMocks[0] },
  user: { ...userMocks[0] },
};

const builder = new ComponentBuilder(
  AccountEditView,
  defaultProps,
);

describe('AccountEditView', () => {
  let component;

  test('matches the snapshot', () => {
    component = builder.shallowGetComponent();
    expect(component).toMatchSnapshot();
  });

  test('renders correctly', () => {
    component = builder.mountComponent();
    expect(component.find('.bookly-account-edit-view')).toHaveLength(1);
    // expect(component.find(Card)).toHaveLength(1);
  });
});
