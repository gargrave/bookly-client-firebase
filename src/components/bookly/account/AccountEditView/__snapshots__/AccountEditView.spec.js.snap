// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountEditView matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountEditView
    disabled={false}
    errors={
      Object {
        "firstName": "",
        "lastName": "",
      }
    }
    onCancelClick={[Function]}
    onInputChange={[Function]}
    onSubmit={[Function]}
    profile={
      Object {
        "created": 2017-07-01T01:16:43.881Z,
        "firstName": "Larry",
        "lastName": "Finklestein",
        "owner": "oiw223h3qndlksadffl3ksdf",
        "updated": 2017-07-01T01:16:43.881Z,
      }
    }
    submitDisabled={false}
    topLevelError=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        classes={
          Array [
            "detail-card",
            "account-edit-card",
          ]
        }
        header="Update My Account"
        hoverable={false}
      >
        <AccountForm
          disabled={false}
          errors={
            Object {
              "firstName": "",
              "lastName": "",
            }
          }
          onCancel={[Function]}
          onInputChange={[Function]}
          onSubmit={[Function]}
          profile={
            Object {
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Larry",
              "lastName": "Finklestein",
              "owner": "oiw223h3qndlksadffl3ksdf",
              "updated": 2017-07-01T01:16:43.881Z,
            }
          }
          submitDisabled={false}
          topLevelError=""
        />
      </Card>,
      "className": "bookly-detail-view bookly-account-edit-view",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <AccountForm
          disabled={false}
          errors={
            Object {
              "firstName": "",
              "lastName": "",
            }
          }
          onCancel={[Function]}
          onInputChange={[Function]}
          onSubmit={[Function]}
          profile={
            Object {
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Larry",
              "lastName": "Finklestein",
              "owner": "oiw223h3qndlksadffl3ksdf",
              "updated": 2017-07-01T01:16:43.881Z,
            }
          }
          submitDisabled={false}
          topLevelError=""
        />,
        "classes": Array [
          "detail-card",
          "account-edit-card",
        ],
        "header": "Update My Account",
        "hoverable": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": false,
          "errors": Object {
            "firstName": "",
            "lastName": "",
          },
          "onCancel": [Function],
          "onInputChange": [Function],
          "onSubmit": [Function],
          "profile": Object {
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Larry",
            "lastName": "Finklestein",
            "owner": "oiw223h3qndlksadffl3ksdf",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "submitDisabled": false,
          "topLevelError": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          classes={
            Array [
              "detail-card",
              "account-edit-card",
            ]
          }
          header="Update My Account"
          hoverable={false}
        >
          <AccountForm
            disabled={false}
            errors={
              Object {
                "firstName": "",
                "lastName": "",
              }
            }
            onCancel={[Function]}
            onInputChange={[Function]}
            onSubmit={[Function]}
            profile={
              Object {
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Larry",
                "lastName": "Finklestein",
                "owner": "oiw223h3qndlksadffl3ksdf",
                "updated": 2017-07-01T01:16:43.881Z,
              }
            }
            submitDisabled={false}
            topLevelError=""
          />
        </Card>,
        "className": "bookly-detail-view bookly-account-edit-view",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <AccountForm
            disabled={false}
            errors={
              Object {
                "firstName": "",
                "lastName": "",
              }
            }
            onCancel={[Function]}
            onInputChange={[Function]}
            onSubmit={[Function]}
            profile={
              Object {
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Larry",
                "lastName": "Finklestein",
                "owner": "oiw223h3qndlksadffl3ksdf",
                "updated": 2017-07-01T01:16:43.881Z,
              }
            }
            submitDisabled={false}
            topLevelError=""
          />,
          "classes": Array [
            "detail-card",
            "account-edit-card",
          ],
          "header": "Update My Account",
          "hoverable": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "errors": Object {
              "firstName": "",
              "lastName": "",
            },
            "onCancel": [Function],
            "onInputChange": [Function],
            "onSubmit": [Function],
            "profile": Object {
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Larry",
              "lastName": "Finklestein",
              "owner": "oiw223h3qndlksadffl3ksdf",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "submitDisabled": false,
            "topLevelError": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
